main.tf

```diff
@@ -1 +1,78 @@
+provider "aws" {}
 
+provider "aws" {
+  alias = "master"
+}
+
+data "aws_caller_identity" "this" {
+}
+
+data "aws_caller_identity" "master" {
+  provider = aws.master
+}
+
+# policy document for MacieHandshake role
+data "aws_iam_policy_document" "this" {
+  count = var.create_macie_member ? 1 : 0
+
+  statement {
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["macie.amazonaws.com"]
+    }
+
+    condition {
+      test     = "StringEquals"
+      variable = "sts:ExternalId"
+
+      values = [
+        data.aws_caller_identity.master.account_id
+      ]
+    }
+  }
+}
+
+# create the MacieHandshake role
+resource "aws_iam_role" "this" {
+  count = var.create_macie_member ? 1 : 0
+
+  name               = "AWSMacieServiceCustomerHandshakeRole"
+  description        = "Allows the master account to create service-linked roles in the member accounts"
+  assume_role_policy = data.aws_iam_policy_document.this[0].json
+  tags               = var.tags
+}
+
+resource "aws_iam_role_policy_attachment" "this" {
+  count = var.create_macie_member ? 1 : 0
+
+  role       = aws_iam_role.this[0].name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonMacieHandshakeRole"
+}
+
+resource "null_resource" "this" {
+  count = var.create_macie_member ? 1 : 0
+
+  # Creation of the MacieHandshake role sometimes takes a few seconds to register,
+  # before the parent account can assume the role, so we pause for a few seconds
+  provisioner "local-exec" {
```

Comment:
_should_ be able to put this provisioner with the sleep in the `aws_iam_role_policy_attachment` resource, instead of having a separate null_resource... then update the depends_on in the `aws_macie_member_account_association` to list just the policy attachment. a bit cleaner that way..

Reply:
Didn't realize that I could put a provisioner in any resource. Updated and the tests are still passing.